src/ring/adapter/jetty9.clj

```diff
@@ -44,39 +45,65 @@
     (string? response) {:body response}
     :else response))
 
+(defn- websocket-upgrade-response? [{:keys [status headers]}]
+  ;; HTTP 101 Switching Protocols
+  ;; https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/101
+  (and (= 101 status)
+       (let [headers (lower-case-keys headers)]
```

Comment:
I wonder if we really need to convert all header names to lower case. This function will be executed on all requests so we'd better keep it as efficient as possible.

Reply:
hmmm... According to https://github.com/ring-clojure/ring/issues/311 we need to deal with the case-issue we accessing headers.

https://github.com/sunng87/ring-jetty9-adapter/blob/81d2b0be1711c5fd798f36cd44f70daf37cef01a/src/ring/adapter/jetty9/common.clj#L18 We have some other case-sensitive header access in this library.

Reply:
> I wonder if we really need to convert all header names to lower case. This function will be executed on all requests so we'd better keep it as efficient as possible.

I'm glad you brought that up because I was hesitant of converting header names at first. The good news is that the `let` form doesn't get evaluated unless the status is 101, which I hope happens less frequently than regular HTTP calls. But anyways, not converting to lower case would be a lot better.

> We have some other case-sensitive header access in this library.

Maybe we could follow that pattern across the board, i.e. case-sensitive, Pascal-kebab case. That means `ws-upgrade-response` would also return header names in that case pattern. What do you think?

Reply:
`(= 101 status)` does great help here. I suggest to follow the http and ring protocol to keep it case insensitive. I will modify prior issues in separated pull request.